presto-main/src/main/java/io/prestosql/sql/planner/plan/GroupIdNode.java

```diff
@@ -67,6 +67,9 @@ public GroupIdNode(
         this.aggregationArguments = ImmutableList.copyOf(aggregationArguments);
         this.groupIdSymbol = requireNonNull(groupIdSymbol);
 
+        checkArgument(source.getOutputSymbols().containsAll(groupingColumns.values()));
+        checkArgument(source.getOutputSymbols().containsAll(aggregationArguments));
+
```

Comment:
I prefer checks in the nodes. It's easier to spot the error where it happens instead of trying to infer what might have caused it after the plane has gone through several transformations. The checkers at the end are good for verifying non-local properties of the plan.

Reply:
`ValidateDependenciesChecker` is last resort check. The best would be to not have `ValidateDependenciesChecker` at all and verify everything along the way.